### ISSUE TITLE
Credo.Check.Refactor.MatchInCondition fails when using pipe operator

### ISSUE THREAD

**Issue by navinpeiris** [rrrene/credo#1228]
In credo version `1.7.14`, `Credo.Check.Refactor.MatchInCondition` fails with the error message "Avoid matches in `if` conditions." when using pipe operator

Example:

```elixir
# this fails with: "Avoid matches in `if` conditions."
if value = map |> Map.get(key) do
  value
else
  default
end

# this is fine
if value = Map.get(map, key) do
  value
else
  default
end
```

**Comment by rrrene** [comment]
Hi, this is sort of the intended behaviour (only allowing super simple assignments).

But I can see that the latest patch breaks existing patterns and we should at least provide an opt-out config param, so this does not trigger an issue. 

Thx for reporting! 🙏

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It should be fixed on `master` with a new config parameter called `allow_operators`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

and configuring the check like this:

```elixir
{Credo.Check.Refactor.MatchInCondition, allow_operators: true}
```

Please report back if your issue is solved! 👍

**Comment by navinpeiris** [comment]
Thanks @rrrene! Can confirm it works perfectly with the new setting 🫶

**Comment by Gladear** [comment]
Hello! I don't know if I should report it in this issue or open a new one, but I've got a similar case that I think should be valid:
```elixir
if token = cookies[@remember_me_cookie] do
```
(this one is generated by [Phoenix scaffolding](https://github.com/phoenixframework/phoenix/blob/07fc5ac215b61849ba54c4cc506dbb1b5248d6d3/priv/templates/phx.gen.auth/auth.ex#L84) 😄)

Let me know what you think!

**Comment by rrrene** [comment]
@Gladear an oversight on my part. Sorry! 😬

Should work on `master`!

**Comment by rrrene** [comment]
This is now available in Credo `1.7.15`! 😄 🎉

**Comment by navinpeiris** [comment]
Updated and works perfectly, thank you @rrrene! 🙏
